assignments/closure.js

```diff
@@ -5,16 +5,48 @@
 // ==== Challenge 2: Create a counter function ====
 const counter = () => {
   // Return a function that when invoked increments and returns a counter variable.
+  let counter = 0;
+  return () => console.log(++counter);
 };
 // Example usage: const newCounter = counter();
 // newCounter(); // 1
 // newCounter(); // 2
 
+const newCounter = counter();
+newCounter();
+newCounter();
+newCounter();
+
 /* STRETCH PROBLEM, Do not attempt until you have completed all previous tasks for today's project files */
 
 // ==== Challenge 3: Create a counter function with an object that can increment and decrement ====
 const counterFactory = () => {
   // Return an object that has two methods called `increment` and `decrement`.
   // `increment` should increment a counter variable in closure scope and return it.
   // `decrement` should decrement the counter variable and return it.
+  let counter = 0;
+  return {
+    increment: function() {
+      console.log(++counter);
+    },
+    decrement: function() {
+      console.log(--counter);
+    }
```

Comment:
You could shorten these with arrow functions; for example:
```javascript
decrement: () => {
  console.log(--counter);
}
```
or even

```javascript
decrement: () => console.log(--counter);
```
although the last could be considered a no-no since you're not necessarily using that value that you're _implicitly_ returning (but it's fine in this case, and JS will let you do it too)